plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/hooks.ts

```diff
@@ -29,6 +29,7 @@ export default ({
   onOverride?: (dataID: string, activeIDs?: string[]) => void;
 }) => {
   const [activeComponentIDs, setActiveIDs] = useState<string[] | undefined>();
+  const latestComponents = useRef<FieldComponent[]>(dataset.components ?? []);
```

Comment:
How is this different from using dataset.components directly?
I think everywhere you are using the latestComponents ref their is dataset in the callback/useEffect's dependencies. You also set latestComponents to dataset.components, so as far as I can tell it'll update in two places, no? 

For example `handleFieldAdd` is called, updates `latestsComponents.current`, then updates the dataset directly. Then, the dataset state gets updated, passes those changes through props back to this hooks file, and then re-updates the useRef. Am I misunderstanding something somewhere?